docker/constraints.txt

```diff
@@ -100,30 +96,50 @@ seaborn==0.11.2
 
 # Geo stack
 cligj==0.7.2
-fiona==1.8.19
+cmocean==2.0
 geohash2==1.1
 geojson==2.5.0
 geopandas==0.9.0
-geopy==2.1.0
-llvmlite==0.36.0
-rasterio==1.2.6
-rasterstats==0.15.0
-requests==2.25.1
-requests-oauthlib==1.3.0
-rio-cogeo==2.3.0
-Shapely==1.7.1
-pyproj==2.6.1.post1
+geopy==2.2.0
 h5py==3.1.0
-netcdf4==1.5.6
-rtree==0.9.7
-owslib==0.24.1
+mapclassify==2.4.3
+OWSLib==0.25.0
 pandana==0.6.1
-pyepsg==0.4.0
+pyproj==3.2.1
 pysal==2.5.0
 pysheds==0.2.7
 python-geohash==0.8.5
-cmocean==2.0
-mapclassify==2.4.3
-rioxarray==0.3.1
 richdem==0.3.4
+rioxarray==0.7.1
+rio-cogeo==2.3.1
+Rtree==0.9.7
 urbanaccess==0.2.2
+
+# nobinary
+aiohttp==3.7.4.post0
+cffi==1.14.6
+cftime==1.5.1
+cryptography==3.4.8
+fiona==1.8.20
+GDAL==3.3.2
+lmdb==1.2.1
+lxml==4.6.3
+msgpack==1.0.2
+multidict==5.1.0
+netcdf4==1.5.7
+protobuf==3.17.3 # due to tensorflow
```

Comment:
`constraints.txt` is meant to be auto-generated with `pip freeze`, so for things like that where you need to work around a version conflict, it's best to list those in `requirements.txt`. Basically `requirements.txt` is a main recipe and contains only necessary constraints *librrary X at least this version but not greater than other", and `constraints.txt` is what gets compiled from `requirements.txt` and then kept around to have reproducible builds.